fix: double announements on the channels

andela-anandaa opened the pull request:
## What's this PR about?
This was being caused by the nested `forEach` loops in bot.js
[here](https://github.com/shaunburdick/slack-ooo/blob/cd6a4e59734f8ee9779d59f37a9f1b2c4126caad/lib/bot.js#L192-L208)

## More info/screenshots, etc.
Example of double messages that were being sent. Sorry I don't know why this didn't catch my eye yesterday -- I guess I was more concerned about the bot posting the message.

<img width="532" alt="screen shot 2016-09-16 at 11 17 52 pm" src="https://cloud.githubusercontent.com/assets/13568167/18599999/db9252e0-7c63-11e6-8e30-c55d3edd9136.png">

<img width="670" alt="screen shot 2016-09-16 at 11 17 39 pm" src="https://cloud.githubusercontent.com/assets/13568167/18600000/db98d962-7c63-11e6-85dd-27eb78a83c28.png">


<img width="755" alt="screen shot 2016-09-16 at 11 16 36 pm" src="https://cloud.githubusercontent.com/assets/13568167/18599960/b02be382-7c63-11e6-9816-c52aeaa2eaae.png">

Also includes a fix **to loop over only valid channels**, for instance I tried this on the config:
```javascript
channels: ['ooo', 'general', 'null']
```
Channel `null` does not exist, so got this on the console log. However, it fails silently as opposed to crashing:
<img width="720" alt="screen shot 2016-09-17 at 1 22 47 am" src="https://cloud.githubusercontent.com/assets/13568167/18603008/8d676e80-7c76-11e6-8b36-e7d210024083.png">

The last fix - dc16343 address this.

andela-anandaa commented:
Fixing the failing tests...

andela-anandaa commented:
Are tests failing due to difference implementation of `Math.round` in Node 5 vs. Node 6? Coz they pass on my local (same to Travis Node 6). Do we change lines 31 - 46 on `bot.test.js` to:

```
// rounding because we are rarely on the minute
  assert.equal(
    Math.ceil(bot.msTillNextTime(times) / 100000), 36,
    'Next time should be about an hour away'
  );

  times.splice(4, 1); // remove one hour time
  assert.equal(
    Math.ceil(bot.msTillNextTime(times) / 100000), 72,
    'Next time should be about two hours away'
  );

  assert.equal(
    Math.ceil(bot.msTillNextTime(times.slice(0, 4)) / 1000000), 72,
    'Next time should be about 20 hours away'
  );
```

andela-anandaa commented:
I can work on the tests for `bot.js` later, I see it's the only one with low coverage.

Please review the PR and let me know if it's okay, I'll be raising one more minor bug ( #16 ) that I came across too, didn't want to mix up the two. Then I'll head over to the issue #1.

shaunburdick commented on lib/bot.js:
```diff
@@ -188,22 +188,25 @@ class Bot {
-              // only announce on channel or group
-              if (
-                (item.is_channel || item.is_group) && // channel or group
-                self.config.app.announce.channels.indexOf(item.name) !== -1
```
Instead of all these changes, couldn't you just change this line to be: `channel === item.name` ?

andela-anandaa replied:
Got it, sending the update now.

shaunburdick commented:
See what I'm talking about [here](https://github.com/shaunburdick/slack-ooo-personal/blob/master/lib/bot.js#L130-L146)

andela-anandaa commented:
Fixed. Also added some minor styling, hope looks good by you, e.g.:

<img width="334" alt="screen shot 2016-09-19 at 3 01 18 pm" src="https://cloud.githubusercontent.com/assets/13568167/18631412/7edbf74a-7e7a-11e6-9763-57284fade1df.png">

shaunburdick commented:
LGTM, There is a rogue space in there but I can get it.
